index out of range in snmp

panic: runtime error: index out of range

goroutine 6315805 [running]:
github.com/lodastack/alarm-adapter/snmp.FetchTraffic(0xc4203d5390, 0xf, 0xc420408370, 0xc4208d0000, 0x106, 0x200, 0xc4204dcb20, 0xb, 0xc42042bc60, 0x19, ...)
        /data/golang/src/github.com/lodastack/alarm-adapter/snmp/collect.go:99 +0x6a8
github.com/lodastack/alarm-adapter/snmp.traffic(0xc4203d5390, 0xf, 0xc4204dcb20, 0xb, 0xc42042bc60, 0x19, 0xc420125210, 0x5)
        /data/golang/src/github.com/lodastack/alarm-adapter/snmp/collect.go:46 +0x236
github.com/lodastack/alarm-adapter/snmp.(*SnmpWorker).Run.func1()
        /data/golang/src/github.com/lodastack/alarm-adapter/snmp/worker.go:117 +0x6c
github.com/lodastack/alarm-adapter/snmp.(*SnmpWorker).Run(0xc4203119e0)
        /data/golang/src/github.com/lodastack/alarm-adapter/snmp/worker.go:125 +0x149
created by github.com/lodastack/alarm-adapter/snmp.(*SnmpMaster).CreateWorker
        /data/golang/src/github.com/lodastack/alarm-adapter/snmp/worker.go:64 +0x158

--- a/snmp/collect.go
+++ b/snmp/collect.go
@@ -85,7 +85,7 @@ func FetchTraffic(ns string, s *gosnmp.GoSNMP, NetworkInfs []NetworkInf, ip stri
 	// ifHCInOctets
 	in_resp, in_err := s.Walk(OID_INF_TRAFFIC_IN)
 	out_resp, out_err := s.Walk(OID_INF_TRAFFIC_OUT)
-	if in_err == nil && out_err == nil && len(in_resp) == len(NetworkInfs) {
+	if in_err == nil && out_err == nil && len(in_resp) == len(NetworkInfs) && len(out_resp) == len(NetworkInfs) {
 		for i, netinterface := range in_resp {
 			var one NetworkTraffic
 			one.oid = netinterface.Name

